test/emqx_channel_SUITE.erl

```diff
@@ -577,7 +577,32 @@ t_handle_out_unexpected(_) ->
 %%--------------------------------------------------------------------
 
 t_handle_call_kick(_) ->
-    {shutdown, kicked, ok, _Chan} = emqx_channel:handle_call(kick, channel()).
+    Channelv5 = channel(),
+    Channelv4 = v4(Channelv5),
+    {shutdown, kicked, ok, _} = emqx_channel:handle_call(kick, Channelv4),
```

Comment:
Suggestion: maybe check that v4 does not issue the same `DISCONNECT` packet that v5 emits?

Reply:
Sorry, I have not understood your meaning too well🤔. This line already checks that in the MQTT v3.1.1 protocol, a DISCONNECT message is not sent out when the kick event is handled. So the pattern matching assertion on this line is enough?

Reply:
I meant that perhaps asserting that the 4th element in the tuple (currently `_`) could be checked to assert that a disconnect packet is _not_ produced, contrary to what is expected in v5.  For instance, one could always return a disconnect packet, and this would not verify that.

Just a nit.  :wink:

Reply:
Got it. Thanks for pointing out this issue. Let me keep it that way for now, it seems to work well enough😆. If possible, we can figure out how to optimize the way these cases are asserted in the whole suite later.